### PR TITLE
Feature/fga/room list api

### PR DESCRIPTION
**Pull request by ganfra**
This PR makes usage of rust-sdk `RoomListService`.
It also includes : 
- Small rework of `NetworkMonitor` 
- A `SyncService` 
- Auto-restart of `Sync` when background/foreground and Network.Online
- A single `RoomSummaryDataSource`

### CODE REVIEW

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt [hunk 1]
```diff
@@ -94,13 +82,15 @@ class RustMatrixClient constructor(
 
     override val sessionId: UserId = UserId(client.userId())
 
+    private val roomListService = client.roomListService()
```
Comment (bmarty) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:85]: What it the diff between `roomListService` and `roomListServiceWithEncryption`?

File: libraries/core/src/main/kotlin/io/element/android/libraries/core/coroutine/ChildScopeOf.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import kotlinx.coroutines.plus
+
+fun childScopeOf(
+    parentScope: CoroutineScope,
+    dispatcher: CoroutineDispatcher,
+    name: String,
+): CoroutineScope = run {
+    val supervisorJob = SupervisorJob(parent = parentScope.coroutineContext.job)
+    parentScope + dispatcher + supervisorJob + CoroutineName(name)
+}
```
Comment (bmarty) [inline libraries/core/src/main/kotlin/io/element/android/libraries/core/coroutine/ChildScopeOf.kt:33]: Maybe rewrite as a Coroutine extension like:

```
/**
 * Create a child scope of the current scope.
 * The child scope will be cancelled if the parent scope is cancelled.
 * The child scope will be cancelled if an exception is thrown in the parent scope.
 *
 * @param dispatcher the dispatcher to use for this scope.
 * @param name the name of the coroutine.
 */
fun CoroutineScope.childScope(
    dispatcher: CoroutineDispatcher,
    name: String,
): CoroutineScope = run {
    val supervisorJob = SupervisorJob(parent = coroutineContext.job)
    this + dispatcher + supervisorJob + CoroutineName(name)
}
```

**Review by bmarty — APPROVED** [review]
LGTM, tested OK an a small account, and on a big account 👍 

With an empty account, I still have this displayed, but we may handle this later:

image

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt (hunk 1 above)
Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:85]: Actually, we should use `roomListServiceWithEncryption` as long as we don't have a dedicated sync for encryption

**Pull request merged**